packages/app/src/containers/SwatchesGroups.tsx

```diff
@@ -0,0 +1,146 @@
+import Sketch from '@sketch-hq/sketch-file-format-ts';
+import {
+  useSelector,
+  useApplicationState,
+} from '../contexts/ApplicationStateContext';
+import { memo, useMemo, useCallback } from 'react';
+import { GroupIcon, MaskOnIcon } from '@radix-ui/react-icons';
+import styled from 'styled-components';
+import { ListView, Spacer } from 'noya-designsystem';
+import { Selectors } from 'noya-state';
+import useShallowArray from '../hooks/useShallowArray';
+
+const Container = styled.div(({ theme }) => ({
+  height: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+}));
+
+const Header = styled.div(({ theme }) => ({
+  fontWeight: 600,
+}));
+
+const sortArray = (array: any[], key: string) =>
+  [...array].sort((a, b) => {
+    const aName = a[key].toUpperCase();
+    const bName = b[key].toUpperCase();
+
+    return aName > bName ? 1 : aName < bName ? -1 : 0;
+  });
+
+type SwatchGroup = {
+  name: string;
+  swatches: Sketch.Swatch[];
+  children?: SwatchGroup[];
```

Comment:
Does it help to make children optional?

Reply:
I was optional for how a function was working before but now it doesn't need to be👍